File: leetcode2/1easy/이준열/2873.cpp
#include <vector>
#include <iostream>
using namespace std;

class Solution {
public:
    long long maximumTripletValue(vector<int>& nums) {
        long long maxVal = 0;
        size_t len = nums.size();

        for (int i = 0; i < len - 2; i++)
        {
            for (int j = i + 1; j < len - 1; j ++)
            {
                long long gap = nums[i] - nums[j];
                if (gap < 0)
                    continue;
                for (int k = j + 1; k < len; k++)
                {
                    long long cur = gap * nums[k];
                    maxVal = std::max(maxVal, cur);
                }
            }
        }
    
        return maxVal;
    }
};

File: leetcode2/1easy/유시연/2873.py
<<<<<<< HEAD
class Leetcode.이재훈.이재훈.Solution:
=======
class Leetcode.이재훈.Solution:
>>>>>>> ab7f46f73d8f18bcd38c5cf06ecabae64f53a424
    def maximumTripletValue(self, nums: List[int]) -> int:
        answer = 0
        for i in range(len(nums)-2):
            for j in range(i+1,len(nums)-1):
                for k in range(j+1, len(nums)):
                    test = (nums[i]- nums[j])*nums[k]
                    answer = max(answer, test)
        
        return answer
        

        